### ISSUE TITLE
Automating Bounty Program Managements.

### ISSUE THREAD

**Issue by flickz** [opencollective/opencollective#2503]
## Idea
Automate the management of bounty program with a Github App, this will save time and allow the program to scale well.

### Some of the tasks that can be automated

- Notify contributors about new bounties added, this can be through Slack message in the engineering channel or separate channel for bounties.

- Ask bounty issue assignee update/progress of the issue daily / every 2 days. Unassign the contributor after a week of no response.

- Keep track of issue PRs, close the issues with bounty claim instructions after they are merged. Example (https://github.com/opencollective/opencollective/issues/2452#issuecomment-538442856)

- Ask feedback from new contributors after new opening their first PR, find out how their experience was contributing and suggestion on how we can improve the developer experience. Ref #2343 

- Report monthly metrics, like numbers of bounty issues closed, PR made, total amount of bounties paid out for the month, contributors leaderboard, "hunter of the month" 😄 .

## Success metrics
The success metrics for this will be time saved managing the bounty issues and developer contributor engagement rate.

**Comment by Betree** [comment]
> Notify contributors about new bounties added, this can be through Slack message in the engineering channel or separate channel for bounties.

That would be fun! We can probably configure that easily with Zapier.

> Ask bounty issue assignee update/progress of the issue daily / every 2 days. Unassign the contributor after a week of no response.

I like to keep it human. Each issue has different requirement and posting a short message to ask for an update is not expensive.

> Keep track of issue PRs, close the issues with bounty claim instructions after they are merged. Example (#2452 (comment))

We usually close issues right after they're merged. For the instructions, I use a GitHub "saved replies" which I can add on the issue in one click.

> Ask feedback from new contributors after new opening their first PR, find out how their experience was contributing and suggestion on how we can improve the developer experience. Ref #2343

That'd be a good one for adding to the saved reply template. Maybe we can do something like:

```

:trophy: This issue is now completed. 

To claim the bounty, you can either:
- Submit an invoice to https://opencollective.com/engineering/expenses/new (please mention the issue number in the expense's title)
- Or ask for an Open Collective gift card of the same value (to donate to other collectives)

---

For more info about the bounty process and how to format your invoice, please read https://docs.opencollective.com/help/developers/bounties

If you have any feedback on how we could improve the bounty process or how to better onboard new developers please [share it with us](https://slack.opencollective.com/), we're always eager to improve :slightly_smiling_face:

```

>  The success metrics for this will be time saved managing the bounty issues and developer contributor engagement rate.

In my current experience, managing bounties is not time consuming at the moment. What is time consuming is reviewing, providing feedback, help contributors to setup their environment or reproduce...etc Improvements for that will come from better docs, guides, CI tooling.

So I don't see the need for a dedicated app right now.

**Comment by piamancini** [comment]
> What is time consuming is reviewing, providing feedback, help contributors to setup their environment or reproduce.

What could we improve in this area? first two seem hard to automate, etc. But improving environment set-ups and documentation for reproducing issues (like, how do I generate a gift card in staging, how do I send an email, etc) seem doable.

**Comment by piamancini** [comment]
PS: Sharing new bounties via zapier to twitter, slack, the contributors group in github is a great idea.

**Comment by Betree** [comment]
> But improving environment set-ups and documentation for reproducing issues (like, how do I generate a gift card in staging, how do I send an email, etc) seem doable.

Absolutely. This event was great for that, it helps to see where contributors are struggling when they discover the codebase. A process that I try to follow is that whenever contributors ask for "How do I do...", I answer by writing some documentation that I'll send them.

I think we're generally doing good on these topics and hopefully it will only get better with time.

**Comment by someofthethings** [comment]
Could you be competing for HackerOne territory soon?

**Comment by Betree** [comment]
@someofthethings No, this is an internal program for Open Collective that we manage through GitHub. This issue is not about a bug bounty (security) program.

**Comment by someofthethings** [comment]
@Betree next time I will be sure to place appropriate emojis on my jokes 🚀

**Comment by Betree** [comment]
Our bounty program is dormant, we keep it to give us the flexibility to pay external contributors when needed but public bounties hasn't been a fantastic experience for us and we don't want to invest more in it at the moment.
